packages/k8s.contrib.crd/internal/SchemaGenerator.pkl

```diff
@@ -0,0 +1,102 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+module k8s.contrib.crd.internal.SchemaGenerator
+
+import "@syntax/ModuleNode.pkl"
+import "@syntax/TypeNode.pkl"
+import "@syntax/ExpressionNode.pkl"
+import "@syntax/ObjectBodyNode.pkl"
+import "ModuleGenerator.pkl"
+
+baseApiGroup: String
+
+crds: Listing<ModuleGenerator.CRD>
+
+fixed moduleNode: ModuleNode = new {
+  declaration {
+    moduleHeader {
+      name {
+        parts {
+          for (part in baseApiGroup.split(".").reverse()) {
+            new { value = part }
+          }
+          new { value = "k8sSchema" }
+        }
+      }
+    }
+  }
+  properties {
+    new {
+      name { value = "resourceTemplates" }
+      typeAnnotation {
+        type = new TypeNode.DeclaredTypeNode {
+          name { parts { new { value = "Mapping" } } }
+          typeArguments {
+            new TypeNode.DeclaredTypeNode {
+              name { parts { new { value = "String" } } }
+            }
+            new TypeNode.DeclaredTypeNode {
+              name { parts { new { value = "Mapping" } } }
+              typeArguments {
+                new TypeNode.DeclaredTypeNode {
+                  name { parts { new { value = "String" } } }
+                }
+                new TypeNode.BuiltInTypeNode { type = "unknown" }
```

Comment:
This should just be `K8sResource` now.

It used to matter (up until Pkl 0.26) because setting a declared type will execute each import when evaluating the map. But, with the lazy typecheck change, this is no longer a concern.